File: templates/nodejs/beginner/bot.js
const { Telegraf } = require('telegraf');
require('dotenv').config();

function createBot(token) {
  const bot = new Telegraf(token || process.env.TELEGRAM_TOKEN);

  bot.start((ctx) => ctx.reply('Welcome! I am a Node.js bot.'));
  bot.help((ctx) => ctx.reply('Send me a message and I will echo it back.'));

  bot.on('text', (ctx) => {
    ctx.reply(`You said: ${ctx.message.text}`);
  });

  return bot;
}

if (require.main === module) {
  const bot = createBot();
  bot.launch().then(() => {
    console.log('Beginner Node.js bot is running...');
  });

  // Enable graceful stop
  process.once('SIGINT', () => bot.stop('SIGINT'));
  process.once('SIGTERM', () => bot.stop('SIGTERM'));
}

module.exports = { createBot };

File: templates/python/beginner/bot.py
import logging
import os
from telegram import Update
from telegram.ext import ApplicationBuilder, ContextTypes, CommandHandler, MessageHandler, filters
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()
TOKEN = os.getenv("TELEGRAM_TOKEN")

# Enable logging
logging.basicConfig(
    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
    level=logging.INFO
)

async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
    """Sends a message when the command /start is issued."""
    await context.bot.send_message(
        chat_id=update.effective_chat.id,
        text="I'm a bot, please talk to me!"
    )

async def echo(update: Update, context: ContextTypes.DEFAULT_TYPE):
    """Echo the user message."""
    await context.bot.send_message(
        chat_id=update.effective_chat.id,
        text=update.message.text
    )

if __name__ == '__main__':
    if not TOKEN:
        print("Error: TELEGRAM_TOKEN not found in environment variables.")
        exit(1)

    application = ApplicationBuilder().token(TOKEN).build()

    start_handler = CommandHandler('start', start)
    echo_handler = MessageHandler(filters.TEXT & (~filters.COMMAND), echo)

    application.add_handler(start_handler)
    application.add_handler(echo_handler)

    print("Bot is starting...")
    application.run_polling()
